### PR TITLE
[mxfp8 moe training] integrate mxfp8 dim0 cast triton kernel

### PR DESCRIPTION
**Pull request by danielvegamyhre**
## Summary
- Integrate triton kernel for mxfp8 quantization along dim0 into torchao. 
- Improve unit tests for mxfp8 kernel and torchao _scaled grouped_mm
- Clean up a unit test for fp8 rowwise (unrelated, but while I was in there)

## Context
There's a torchtitan issue triggered by torch.compile in the MoE layer where the all to all input_splits don't match the size of the data. So for now, we need to be able to run the MoE layer in eager mode with good performance.

Also there have been 2 torch.compile perf regressions for this quantization op in the past couple weeks that were tricky to deal with, so it will be useful to have a reliable triton kernel we can for now to circumvent all of this. (We are working together with compile folks on getting some unit tests in core to prevent future regressions)

## Test plan
- Unit tests:
    - `pytest test/prototype/mx_formats/test_kernels.py -k dim0`
    - `pytest test/prototype/moe_training/test_scaled_grouped_mm.py -k mxfp8`
- torchtitan e2e test:  1000 step run on 8xB200, validated no NaNs and good loss curve
    - `LOG_RANK=4 NGPU=8 CONFIG_FILE="/home/${USER}/torchtitan/torchtitan/models/llama4/train_configs/llama4_17bx16e.toml" ./run_train.sh --metrics.log_freq=10 --training.steps=1000  --parallelism.data_parallel_shard_degree=4 --parallelism.expert_parallel_degree=4 --parallelism.tensor_parallel_degree=1 --parallelism.expert_tensor_parallel_degree=1 --training.seq_len=8192 --training.local_batch_size=12 --model.print_after_conversion --training.debug_moe_force_load_balance --activation_checkpoint.mode="full" --parallelism.pipeline_parallel_degree 2 --parallelism.pipeline_parallel_schedule "Interleaved1F1B" --parallelism.pipeline_parallel_layers_per_stage 1`

```
[rank4]:[titan] 2025-10-16 10:34:46,773 - root - INFO - step: 1000  loss:  5.0545  grad_norm:  0.2730  memory: 118.22GiB(66.28%)  tps: 30,005  tflops: 329.75  mfu: 14.66%
```

### CODE REVIEW

File: test/prototype/mx_formats/test_kernels.py [hunk 1]
```diff
@@ -499,6 +499,21 @@ def test_triton_mxfp8_dim0_randn(M, K):
...
+@pytest.mark.skipif(not has_triton(), reason="unsupported without triton")
+@pytest.mark.skipif(
+    not is_sm_at_least_100(),
+    reason="mxfp8 requires CUDA capability 10.0 or greater",
+)
+def test_triton_mxfp8_dim0_zeros():
+    x = torch.zeros(8192, 5120, dtype=torch.bfloat16, device="cuda")
+    x_mx_ref, x_s_ref = triton_to_mxfp8_dim0_reference(x, block_size=32)
+    x_mx_t, x_s_t = triton_to_mxfp8_dim0(x, inner_block_size=32)
+    breakpoint()
```
Comment (vkuzo) [inline test/prototype/mx_formats/test_kernels.py]: remove

Comment (danielvegamyhre) [reply test/prototype/mx_formats/test_kernels.py]: Woops, thanks

**Comment by danielvegamyhre** [comment]
failing nf4 test unrelated
